### PR TITLE
Update to go1.11.1

### LINKED ISSUE #67872: Update to go1.11
Keeping 👀on this. Official [docker upstream image](https://hub.docker.com/r/library/golang/) isn't available yet, so I can't build kube-cross locally yet.

Will need gofmt changes for sure (and they aren't backwards compatible).

cc: @ixdy @luxas

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Updates kubernetes to build under go1.11.1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #67872

**Special notes for your reviewer**:
Will need CI images/environments to be bumped too, as there are gofmt changes.

**Release note**:
```release-note
Update to use go1.11.1
```

/assign @ixdy @BenTheElder 
/cc @dims @kubernetes/sig-release-misc @kubernetes/sig-scalability-misc

### CODE REVIEW

**Comment by cblecker** [comment]
oh, also cc: @kubernetes/sig-testing regarding CI bumps

**Comment by timothysc** [comment]
/assign @shyamjvs 

@kubernetes/sig-scalability-feature-requests typically owns the 👍 👎  on go versions.

**Review by BenTheElder — APPROVED** [review]
/lgtm
:tada:

File: build/root/WORKSPACE
Comment (ixdy) [inline build/root/WORKSPACE]: rules_go 0.15.4 is out now with support for go1.11.1.

**Comment by BenTheElder** [comment]
working on the CI image bumps

File: build/root/WORKSPACE
Comment (cblecker) [reply build/root/WORKSPACE]: Updated!

File: build/build-image/cross/VERSION
Comment (ixdy) [inline build/build-image/cross/VERSION]: I've built and pushed k8s.gcr.io/kube-cross:v1.11.1-1. Waiting for it to sync to mirrors.

File: hack/lib/golang.sh
Comment (ixdy) [inline hack/lib/golang.sh]: hm, lots of verify checks fail unless verify is bumped first, but bumping verify first means that any other PR will fail (due to gofmt failures).

**Comment by BenTheElder** [comment]
note: verify is going to fail in some capacity once we merge the revert in https://github.com/kubernetes/test-infra/pull/9691

some of the failures look legitimate :(

**Comment by ixdy** [comment]
The bazel-test failure is a bit of a mess. https://github.com/kubernetes/repo-infra/issues/86

**Comment by cblecker** [comment]
The current bazel test failure may be a bazel specific rules_go issue.
- bazel log from CI (`bazel test //staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope:go_default_test
`):
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69386/pull-kubernetes-bazel-test/62546/#stagingsrck8sioapiserverpkgstoragevalueencryptenvelopego_default_test
- my local output from go-native 1.11 testing:
```
$ KUBE_RUN_COPY_OUTPUT=n build/run.sh go test -race -v ./staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope
+++ [1004 15:47:59] Verifying Prerequisites....
+++ [1004 15:47:59] Using Docker for MacOS
+++ [1004 15:47:59] Building Docker image kube-build:build-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:32] Keeping image kube-build:build-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:32] Creating data container kube-build-data-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:39] Syncing sources to container
+++ [1004 15:49:40] Stopping any currently running rsyncd container
+++ [1004 15:49:40] Starting rsyncd container
+++ [1004 15:49:41] Running rsync
+++ [1004 15:49:55] Stopping any currently running rsyncd container
+++ [1004 15:49:56] Output from this container will NOT be rsynced out upon completion. Set KUBE_RUN_COPY_OUTPUT=y to enable.
+++ [1004 15:49:56] Running build command...
=== RUN   TestEnvelopeCaching
--- PASS: TestEnvelopeCaching (0.00s)
=== RUN   TestEnvelopeCacheLimit
--- PASS: TestEnvelopeCacheLimit (0.00s)
=== RUN   TestBackwardsCompatibility
--- PASS: TestBackwardsCompatibility (0.00s)
=== RUN   TestKMSPluginLateStart
E1004 15:50:05.553044    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
E1004 15:50:06.554492    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
--- PASS: TestKMSPluginLateStart (2.68s)
=== RUN   TestIntermittentConnectionLoss
E1004 15:50:08.235518    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
--- PASS: TestIntermittentConnectionLoss (1.01s)
    grpc_service_unix_test.go:93: Connected to KMSPlugin
    grpc_service_unix_test.go:97: KMS Plugin is stopped
    grpc_service_unix_test.go:104: Sending encrypt request
    grpc_service_unix_test.go:120: Restarted KMS Plugin
=== RUN   TestUnsupportedVersion
--- PASS: TestUnsupportedVersion (0.01s)
=== RUN   TestConcurrentAccess
--- PASS: TestConcurrentAccess (0.00s)
=== RUN   TestGRPCService
--- PASS: TestGRPCService (0.01s)
=== RUN   TestGRPCServiceConcurrentAccess
--- PASS: TestGRPCServiceConcurrentAccess (1.35s)
=== RUN   TestInvalidConfiguration
=== RUN   TestInvalidConfiguration/emptyConfiguration
=== RUN   TestInvalidConfiguration/invalidScheme
--- PASS: TestInvalidConfiguration (0.00s)
    --- PASS: TestInvalidConfiguration/emptyConfiguration (0.00s)
    --- PASS: TestInvalidConfiguration/invalidScheme (0.00s)
PASS
ok      k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope       6.124s
```

cc: @ixdy

**Comment by ixdy** [comment]
hm, the second bazel-test invocation passed. Could it maybe be the test? It was only added about a week ago in #68585.

(OTOH, it isn't showing any failures in http://k8s-testgrid.appspot.com/sig-release-master-blocking#bazel-test&sort-by-flakiness=&width=5.)

cc @immutableT

**Comment by BenTheElder** [comment]
That test uses `sleep` which immediately sets off my flake-radar :^)

**Comment by ixdy** [comment]
Ah, good sleuthing. Also, the CI bazel-test job may not be the best metric, since I think a lot of tests end up being cached there.

Looking at [presubmit bazel-test](http://k8s-testgrid.appspot.com/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&sort-by-flakiness=), this is the third most flakiest test, with similar failures as we saw here. So I don't think this is a go1.11.1 issue.

**Comment by immutableT** [comment]
It may be that envelope test is a bit too ambitions - trying to simulate scenarios of recovering from  intermittent network failures, and it is timing dependent. I will take a look at it in the morning.

**Comment by immutableT** [comment]
69470 increases time-out in staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope; this should result in the reduction of intermittent failures. I will continue monitoring this test.

**Comment by lavalamp** [comment]
Happy to approve once the tests look a bit greener :)

**Comment by ixdy** [comment]
I'm not sure exactly how to proceed here.

Most of the test failures are because the minimum go version check requires go1.11.1.

We could punt that to a separate PR, though the verify job will probably still fail, due to `gofmt` changes.

If we bump the images to go1.11.1, then this PR will pass, but the verify job will fail all other PRs in master for a bit.

**Comment by cblecker** [comment]
Because the fmt changes and the TestGetNameFromCallsite fix both require go1.11, I think they should all stay in the same PR.

I think we just need to pick a time, bump the test-infra images, and then retest this for merge.

**Comment by ixdy** [comment]
@lavalamp would this^ proposal work for you?

**Comment by lavalamp** [comment]
Yes, that works for me as long as we warn people (message the dev mailing
list?) and we're pretty confident that we won't find some other test
failure that's currently masked.

**Comment by cblecker** [comment]
Rebased again to resolve conflicts with the gofmt stuff

**Comment by BenTheElder** [comment]
I'm ready to switch the images back to go1.11 from the infra side any time, please let me know. IMO we should get this done. I think we should mail the ML _after_ we switch to go1.11 and managed to merge this, otherwise we'll just be spamming them when we have to revert.

**Comment by BenTheElder** [comment]
We also should not find any other failures, a lot of the tests use bazel which will switch to go1.11 without any image changes, out of the jobs that are failing:
- verify is failing due to not having 1.11. we switched breifly the other day to 1.11 on accident, and it was fine
- typecheck is failing due to the go1.11 version check
- local-e2e-containerized is generically broken currently and does not block PRs
- node-e2e is failing the go1.11 version check
- integration is failing the version check, and is eternally flaky so we can't get a good signal anyhow :man_shrugging:

**Comment by cblecker** [comment]
SGTM.

@lavalamp: Can we get an approve then, based on the above?

**Comment by BenTheElder** [comment]
Image bump is ready in https://github.com/kubernetes/test-infra/pull/9716

**Comment by lavalamp** [comment]
/approve

/hold

unhold when ready!

**Comment by lavalamp** [comment]
also

/lgtm

**Comment by lavalamp** [comment]
/hold cancel

I guess that wasn't really necessary.

**Comment by lavalamp** [comment]
I sent a heads-up to the kubernetes-dev mailing list, since this will be disruptive.

**Comment by BenTheElder** [comment]
I think integration may have flaked, but that's kinda to be expected with integration lately :/

**Comment by lavalamp** [comment]
@cblecker @BenTheElder looks like an actual go vet failure.

Fix asap, merge while broken & quick follow-up, or abandon ship?

**Comment by BenTheElder** [comment]
fix asap, if we abandon ship and go back to go1.10 more of these will be introduced, these are recent changes in master that don't work for 1.11

**Comment by BenTheElder** [comment]
they also look like very trivial fixes hopefully. they're all in _our_ code, and they're small stuff like printf mistakes

**Comment by lavalamp** [comment]
OK, we'll wait for a fix in this PR then :)

File: test/soak/serve_hostnames/serve_hostnames.go
Comment (BenTheElder) [inline test/soak/serve_hostnames/serve_hostnames.go]: this doesn't seem right?

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Pull request merged**
